src/components/overflow-menu-drawer/TrackOverflowMenuDrawer.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react'
+
+import {
+  FavoriteSource,
+  FollowSource,
+  RepostSource,
+  ShareSource
+} from 'audius-client/src/common/models/Analytics'
+import { ID } from 'audius-client/src/common/models/Identifiers'
+import { CommonState } from 'audius-client/src/common/store'
+import { getTrack } from 'audius-client/src/common/store/cache/tracks/selectors'
+import { getUser } from 'audius-client/src/common/store/cache/users/selectors'
+// Importing directly from audius-client for now, this will be removed
+// when the profile page is implemented in RN
+import {
+  repostTrack,
+  undoRepostTrack,
+  saveTrack,
+  unsaveTrack,
+  shareTrack
+} from 'audius-client/src/common/store/social/tracks/actions'
+import {
+  followUser,
+  unfollowUser
+} from 'audius-client/src/common/store/social/users/actions'
+import { requestOpen as openAddToPlaylistModal } from 'audius-client/src/common/store/ui/add-to-playlist/actions'
+import { getMobileOverflowModal } from 'audius-client/src/common/store/ui/mobile-overflow-menu/selectors'
+import {
+  OverflowAction,
+  OverflowActionCallbacks
+} from 'audius-client/src/common/store/ui/mobile-overflow-menu/types'
+import { requestOpen as openTikTokModal } from 'audius-client/src/common/store/ui/share-sound-to-tiktok-modal/slice'
+import { profilePage } from 'audius-client/src/utils/route'
+import { push } from 'connected-react-router'
+
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { useSelectorWeb } from 'app/hooks/useSelectorWeb'
+
+type Props = {
+  render: (callbacks: OverflowActionCallbacks) => React.ReactNode
+}
+
+const TrackOverflowMenuDrawer = ({ render }: Props) => {
+  const dispatchWeb = useDispatchWeb()
+  const { id: modalId } = useSelectorWeb(getMobileOverflowModal)
+  const id = modalId as ID
+
+  const { owner_id, title, permalink } = useSelectorWeb((state: CommonState) =>
+    getTrack(state, { id })
+  )
+
+  const { handle } = useSelectorWeb((state: CommonState) =>
+    getUser(state, { id: owner_id })
+  )
+
+  if (!id || !owner_id || !handle || !title) {
+    return <></>
+  }
+
+  const callbacks = {
```

Comment:
i like this pattern. is it worth a useMemo?

Reply:
I don't think it's worth it, because this is near the top of the React tree this should only ever rerender when the `mobileOverflowModal` state changes. In which case these callbacks need to be redefined anyway